alphacsc/utils/coordinate_descent.py

```diff
@@ -54,17 +54,18 @@ def _coordinate_descent_idx(Xi, D, constants, reg, z0=None, max_iter=1000,
     else:
         z_hat = z0.copy()
 
+    n_times_seg = 2 * np.array(n_times_atom) - 1
     if n_seg == 'auto':
         if strategy == 'greedy':
-            n_seg = max(n_times_valid // (2 * n_times_atom), 1)
+            n_seg = max(1, n_times_valid // n_times_seg +
+                        ((n_times_valid % n_times_seg) != 0))
         elif strategy in ('random', 'cyclic'):
             n_seg = 1
             n_coordinates = n_times_valid * n_atoms
 
```

Comment:
This code will break if `n_seg` is not `'auto'`, as `n_times_seg` won't be correct.
Could you add a `else` that compute `n_times_seg` when `n_seg` is given by the user:
```suggestion
    else:
        n_times_seg = n_times_valid // n_seg + (n_times_valid % n_seg != 0)
```